src/components/tools/Partials/ChangeLocation.svelte

```diff
@@ -69,9 +70,23 @@
 
   async function updateBoundary(e) {
     if (!e.detail) return;
+    const { lng, lat } = currentLoc.center;
     currentBoundary = e.detail;
     searchPlaceholder = `Enter ${currentBoundary.metadata.placeholder}`;
-    currentLoc = await getFeature(currentLoc, currentBoundary.id);
+    const intersectingFeature = await getFeature(
+      currentLoc,
+      currentBoundary.id
+    );
+    if (intersectingFeature) {
+      currentLoc = intersectingFeature;
+    } else {
+      const nearestFeature = await getNearestFeature(
```

Comment:
Could `getNearestFeature` potentially return an empty value or fail?

Reply:
`getNearestFeature` should not return an empty value, but it could fail. I'll add more error handling as part of another PR where I fix an issue with the state boundary.

Reply:
On second thoughts I'll add the error handling to this PR instead of merging it just yet.